ODT/ODTUtility.php

```diff
@@ -194,47 +194,62 @@ public static function deleteUselessElements(&$docContent, array $preventDeletet
      *                Just the integer value, no units included.
      */
     public static function getImageSize($src, $maxwidth=NULL, $maxheight=NULL){
-        if (file_exists($src)) {
-            $info  = getimagesize($src);
-            if(!$info)
-            {
-                $svgfile = simplexml_load_file($src);
-                if(isset($svgfile["width"]) && isset($svgfile["height"]))
-                {
-                    $info = array(substr($svgfile["width"],0,-2), substr($svgfile["height"],0,-2));
-                }
-                elseif (isset($svgfile["viewBox"]))
-                {
-                    /* preg_match("#viewbox=[\"']\d* \d* (\d*+(\.?+\d*)) (\d*+(\.?+\d*))#i", file_get_contents($src), $info);
-                    $info = array($info[1], $info[3]); */
-                    $info = explode(' ', $svgfile["viewBox"]);
-                    $info = array($info[2], $info[3]);
-                }
+        if(file_exists($src)) {
+            $info = getimagesize($src);
+        } else {
+            $fetch = (new DokuHTTPClient())->get($src);
```

Comment:
If a file is not found, then it tries to download again? (and I guess time out again? Unless there was a proxy involved?) if this is needed for proxies, can it not be recognized before trying? Doing downloads is one of the slower activities we could let do our webserver. So it does not look so optimal to try it twice?

Reply:
This line of code is better explained through the plantumlparser plugin: 

https://github.com/eduardomozart/dokuwiki_plantumlparser/blob/master/syntax/injector.php#L112

The PlantUML plugin uses the PlantUML Online Server (e.g. http://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa70000) to host the image, so creating the following diagram at your Wiki:

```
<uml>
@startuml
Bob -> Alice : hello
@enduml
</uml>
```

It embeds it's SVG file on the page directly from PlantUML Web Server. The resultant URL of the PlantUML syntax above is: 

```
http://www.plantuml.com/plantuml/svg/SoWkIImgAStDuNBAJrBGjLDmpCbCJbMmKiX8pSd9vt98pKi1IW80
```

This piece of code to download image also applies to external images, e.g.:

```
http://www.plantuml.com/plantuml/png/SoWkIImgAStDuNBAJrBGjLDmpCbCJbMmKiX8pSd9vt98pKi1IW80
```

The ``getimagesize`` PHP function allows to parse remote image files, but if ``allow_url_fopen`` PHP directive is off or if the server is behind a proxy, this download may fail, so it attempts to use the DokuWiki built-in HTTP client to download the file (assuming that if it doesn't exist locally it's a remote file) and only if the download fails the image is considered unreachable.